opensrp-chw/src/main/java/org/smartregister/chw/fragment/ReferralJsonWizardFormFragment.java

```diff
@@ -24,9 +27,34 @@ public static ReferralJsonWizardFormFragment getFormFragment(String stepName) {
 
     @Override
     public void customClick(Context context, String behaviour){
-        Toast.makeText(context, getResources().getString(R.string.referral_submitted), Toast.LENGTH_LONG).show();
-        //save
-        save();
+        // check if referral exist
+        String businessStatus = behaviour.equalsIgnoreCase("refer") ? CoreConstants.BUSINESS_STATUS.REFERRED : CoreConstants.BUSINESS_STATUS.LINKED;
+        String baseEntityID = ((ReferralWizardFormActivity) getActivity()).getBaseEntityID();
+        String referredTo = behaviour.equalsIgnoreCase("refer") ? "Health Facility" : "Addo";
+
+        if (CoreReferralUtils.hasReferralTask(baseEntityID, businessStatus) && !behaviour.equalsIgnoreCase("save")) {
+            AlertDialog dialog = new AlertDialog.Builder(getActivity())
+                    .setMessage("This client already has a referral to the "+ referredTo +", do you want to close this referral and open a new one?")
+                    .setTitle("Existing Referral")
```

Comment:
So for translations we should not forget to add these to the strings

Reply:
yes